nilearn/externals/skimage/random_walker_segmentation.py

```diff
@@ -0,0 +1,516 @@
+
+    Examples
+    --------
+    >>> np.random.seed(0)
+    >>> a = np.zeros((10, 10)) + 0.2 * np.random.rand(10, 10)
+    >>> a[5:8, 5:8] += 1
+    >>> b = np.zeros_like(a)
+    >>> b[3, 3] = 1  # Marker for first phase
+    >>> b[6, 6] = 2  # Marker for second phase
+    >>> random_walker(a, b)
+    array([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+           [1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+           [1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+           [1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+           [1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+           [1, 1, 1, 1, 1, 2, 2, 2, 1, 1],
+           [1, 1, 1, 1, 1, 2, 2, 2, 1, 1],
+           [1, 1, 1, 1, 1, 2, 2, 2, 1, 1],
+           [1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+           [1, 1, 1, 1, 1, 1, 1, 1, 1, 1]], dtype=int32)
+
+    """
+    # Parse input data
+    if mode is None:
+        if amg_loaded:
+            mode = 'cg_mg'
+        elif UmfpackContext is not None:
+            mode = 'cg'
+        else:
+            mode = 'bf'
+
+    if UmfpackContext is None and mode == 'cg':
+        warnings.warn('"cg" mode will be used, but it may be slower than '
+                      '"bf" because SciPy was built without UMFPACK. Consider'
+                      ' rebuilding SciPy with UMFPACK; this will greatly '
+                      'accelerate the conjugate gradient ("cg") solver. '
+                      'You may also install pyamg and run the random_walker '
+                      'function in "cg_mg" mode (see docstring).')
+
+    if (labels != 0).all():
+        warnings.warn('Random walker only segments unlabeled areas, where '
+                      'labels == 0. No zero valued areas in labels were '
+                      'found. Returning provided labels.')
+
+        if return_full_prob:
+            # Find and iterate over valid labels
+            unique_labels = np.unique(labels)
+            unique_labels = unique_labels[unique_labels > 0]
+
+            out_labels = np.empty(labels.shape + (len(unique_labels),),
+                                  dtype=np.bool)
+            for n, i in enumerate(unique_labels):
+                out_labels[..., n] = (labels == i)
+
+        else:
+            out_labels = labels
+        return out_labels
+
+    # This algorithm expects 4-D arrays of floats, where the first three
+    # dimensions are spatial and the final denotes channels. 2-D images have
+    # a singleton placeholder dimension added for the third spatial dimension,
+    # and single channel images likewise have a singleton added for channels.
+    # The following block ensures valid input and coerces it to the correct
+    # form.
+    if not multichannel:
+        if data.ndim < 2 or data.ndim > 3:
+            raise ValueError('For non-multichannel input, data must be of '
+                             'dimension 2 or 3.')
+        dims = data.shape  # To reshape final labeled result
+        data = np.atleast_3d(img_as_float(data))[..., np.newaxis]
```

Comment:
Ok. I would do this importing from sklearn. Do you think it is Ok for using like this ?

```python
from sklearn.utils import as_float_array
data = as_float_array(data)[..., np.newaxis]
```